### PR TITLE
dev/core#4364 Use writeRecord for Navigations so menu changes for managed entities don't reset

### PR DESCRIPTION
**Pull request by larssandergreen**
Overview
----------------------------------------
See [issue.](https://lab.civicrm.org/dev/core/-/issues/4364)
This resolves issues with updates, but deletions still have the problem described in the issue, pending #24496.

Before
----------------------------------------
Pre and post hooks aren't called when adding, modifying or deleting navigation menu items, so these are reset on cache clear, erasing user changes.

After
----------------------------------------
Pre and post hooks are called.

Technical Details
----------------------------------------
We lose a transaction that used to wrap the weight changes in processMove. This seems fine though as the worst case scenario is the first query succeeds, but the writeRecord fails, in which case we have simply inserted a gap in the menu weights, which doesn't matter.

### CODE REVIEW

File: CRM/Core/BAO/Navigation.php [hunk 1]
```diff
@@ -578,8 +574,6 @@ public static function processMove($nodeID, $referenceID, $position) {
       $incrementOtherNodes = FALSE;
     }
 
-    $transaction = new CRM_Core_Transaction();
-
     // now update the existing nodes to weight + 1, if required.
     if ($incrementOtherNodes) {
       $query = "UPDATE civicrm_navigation SET weight = weight + 1
```
Comment (colemanw) [inline CRM/Core/BAO/Navigation.php:579]: Note this still doesn't call hooks when updating *other* items in the tree. That's fine, but makes https://github.com/civicrm/civicrm-core/pull/27814 still important to merge.

Comment (larssandergreen) [reply CRM/Core/BAO/Navigation.php:579]: Agree, seemed unnecessary to call the hooks for potentially a large number of menu items.

**Comment by larssandergreen** [comment]
Pushed a new version that removes the other two uses of processDelete, in reports.

**Comment by colemanw** [comment]
@larssandergreen given that we've been recently been burned by removing core functions, could you please mark the 2 functions as `@deprecated` instead of deleting them (and switch their guts to just emit a deprecated notice and call `writeRecord()`)

**Comment by larssandergreen** [comment]
Yes, I just grepped universe and found a bunch of these, was obviously too optimistic with this one.

**Comment by colemanw** [comment]
That's pretty bad considering that there *is* an api for extensions to use [since all the way back in 2015](https://github.com/civicrm/civicrm-core/commit/3523b615bc4e212d226605c25373c20efc365f3c), but it's a wild-wild `universe` out there...

**Comment by colemanw** [comment]
@larssandergreen I've removed `merge-on-pass`... once you reinstate those functions we can merge this!

**Pull request merged**